tensorflow/contrib/ignite/kernels/ignite_dataset_iterator.h

```diff
@@ -0,0 +1,77 @@
+/* Copyright 2018 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+#include "ignite_binary_object_parser.h"
+#include "ignite_client.h"
+#include "ignite_dataset.h"
+
+namespace tensorflow {
+
+class IgniteDatasetIterator : public DatasetIterator<IgniteDataset> {
+ public:
+  IgniteDatasetIterator(const Params& params, std::string host, int32 port,
+                        std::string cache_name, bool local, int32 part,
+                        int32 page_size, std::string username,
+                        std::string password, std::string certfile,
+                        std::string keyfile, std::string cert_password,
+                        std::vector<int32> schema,
+                        std::vector<int32> permutation);
+  ~IgniteDatasetIterator();
+  Status GetNextInternal(IteratorContext* ctx, std::vector<Tensor>* out_tensors,
+                         bool* end_of_sequence) override;
+
+ protected:
+  Status SaveInternal(IteratorStateWriter* writer) override;
+  Status RestoreInternal(IteratorContext* ctx,
+                         IteratorStateReader* reader) override;
+
+ private:
```

Comment:
Style nit: in classes, within each section (`public`, `private` etc.), [put method declarations before data member declarations](https://google.github.io/styleguide/cppguide.html#Declaration_Order).

Reply:
Done.